### PR TITLE
Support patching objects provided by path

### PR DESCRIPTION
**Pull request by graingert**
For when you are too lazy to even import the function you want to patch

### CODE REVIEW

**Comment by adamchainz** [comment]
Is the use case really just to avoid the import statement? That seems a bit too lazy to me. Writing `import` is just one line, and it avoids patchy having to touch the import machinery, which is always more complicated than it seems. I'm pretty sure this patch doesn't cover *every* case of how `import` works.

**Comment by graingert** [comment]
The import machinery is lifted directly from mock

File: tests/test_patch.py [hunk 1]
```diff
@@ -751,3 +751,23 @@ def sample():
...
+                return 1
+        """))
+    sys.path.insert(0, six.text_type(tmpdir))
+    try:
+        patchy.patch('patch_by_path.Foo.sample', """\
+            @@ -2,2 +2,2 @@
+            -    return 1
+            +    return 2
+            """)
+        from patch_by_path import Foo
```
Comment (adamchainz) [inline tests/test_patch.py]: Worth testing both cases where the caller has and has not already imported Foo?

**Comment by adamchainz** [comment]
This needs rebasing now #155 has been merged :)

File: tests/test_patch.py (hunk 1 above)
Comment (graingert) [reply tests/test_patch.py]: done

**Comment by graingert** [comment]
@adamchainz rebased

**Comment by adamchainz** [comment]
Realized I can push to your branch :)

**Pull request merged**